### PR TITLE
Setup varnish for caching requests going to content and lists endpoints

### PR DESCRIPTION
**Pull request by tosan88**
Finally we managed to setup varnish as default entry point for requests in xp

### CODE REVIEW

File: varnish-sidekick@.service [hunk 1]
```diff
@@ -5,15 +5,17 @@ After=varnish@%i.service
...
-    etcdctl set /services/varnish/%i/host $VARNISH_IP --ttl 60; \
-    etcdctl set /services/varnish/%i/port $VARNISH_PORT --ttl 60; \
+    export PORT=$(/usr/bin/docker port varnish-%i 80 | cut -d':' -f2); \
+    etcdctl set /ft/services/varnish/servers/%i http://$HOSTNAME:$PORT --ttl 600; \
     sleep 45; \
   done"
-ExecStop=/usr/bin/etcdctl rm /services/varnish@%i
+ExecStop=/bin/sh -c "\
+  etcdctl rm /ft/services/varnish/servers/%i; \
+  etcdctl rm /vulcand/frontends/varnish-public/frontend"
```
Comment (BerniVarga) [inline varnish-sidekick@.service]: Does this key still exist?

Comment (tosan88) [inline varnish-sidekick@.service]: No, you're right. I'll fix this

File: varnish-sidekick@.service [hunk 2]
```diff
@@ -5,15 +5,16 @@ After=varnish@%i.service
...
-  export VARNISH_IP=$HOSTNAME; \
+  etcdctl set   /ft/services/varnish/healthcheck true; \
+  etcdctl mkdir /ft/services/varnish/servers; \
+  etcdctl set /ft/healthcheck/varnish-%i /__health; \
   while true; do \
-    export VARNISH_PORT=$(/usr/bin/docker port varnish-%i 80 | cut -d':' -f2); \
-    etcdctl set /services/varnish/%i/host $VARNISH_IP --ttl 60; \
-    etcdctl set /services/varnish/%i/port $VARNISH_PORT --ttl 60; \
+    export PORT=$(/usr/bin/docker port varnish-%i 80 | cut -d':' -f2); \
+    etcdctl set /ft/services/varnish/servers/%i http://$HOSTNAME:$PORT --ttl 600; \
```
Comment (Zapadlo) [inline varnish-sidekick@.service]: If you ttl is 600s you probably don't need to refresh it every 45s

**Comment by Zapadlo** [comment]
I would have still preferred a less chatty approach with linking varnishes to APCs, but..

+1

**Comment by BerniVarga** [comment]
+1 (Would it make sense to put a task of optimization on our board for varnish-apc coupling?)

File: varnish-sidekick@.service (hunk 2 above)
Comment (tosan88) [inline varnish-sidekick@.service]: Ah, so what would fit better: to change TTL to be less (120s maybe) or to increase the sleep? or both? :)

**Comment by tosan88** [comment]
@Zapadlo @BerniVarga I thought the same, we could implement linking as a further improvement

**Comment by Zapadlo** [comment]
@tosan88 sgtm

File: varnish-sidekick@.service (hunk 2 above)
Comment (Zapadlo) [inline varnish-sidekick@.service]: How about ttl of 5 and sleep of 4?

In line with this: https://github.com/Financial-Times/up-service-files/pull/271

Comment (tosan88) [inline varnish-sidekick@.service]: Updated.

**Pull request merged**